numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -398,6 +398,13 @@ _strided_to_strided_contig_align_wrap(char *dst, npy_intp dst_stride,
     npy_bool init_dest = d->init_dest, out_needs_api = d->out_needs_api;
 
     for(;;) {
+        // Our caller does not know if a previous call resulted in a Python
+        // exception. Much of the Python API is unsafe to use while an exception
+        // is in flight, so just skip all of the work. Someone higher in the
+        // call stack will check for errors and propagate them.
```

Comment:
Numpy tends to use C-style comments internally, in the style of:
```suggestion
        /* Our caller does not know if a previous call resulted in a Python
         * exception. Much of the Python API is unsafe to use while an exception
         * is in flight, so just skip all of the work. Someone higher in the
         * call stack will check for errors and propagate them.
         */
```